server/metrics/global_status.go

```diff
@@ -0,0 +1,216 @@
+	}
+}
+
+func (r *RequestStatus) AddWriteBytes(value int64) {
+	if r.countWrites {
+		r.writeBytes += value
+		log.Debug().Int64("Bytes written", value).Int64("Total bytes written", r.writeBytes).Msg("Added written bytes")
+	}
+}
+
+func (r *RequestStatus) SetWriteBytes(value int64) {
+	r.writeBytes = value
+}
+
+func (r *RequestStatus) SetReadBytes(value int64) {
+	r.readBytes = value
+}
+
+func (r *RequestStatus) GetReadBytes() int64 {
+	return r.readBytes
+}
+
+func (r *RequestStatus) GetWriteBytes() int64 {
+	return r.writeBytes
+}
+
+func (g *GlobalStatus) MergeActiveChunk() {
+	endTime := time.Now()
+	g.mu.Lock()
+	activeChunk := *g.activeChunk
+	log.Debug().Time("start time", activeChunk.startTime).Time("end time", endTime).Msg("Merging active chunk")
+	activeChunk.endTime = endTime
+	g.timeChunks[activeChunk.startTime] = &activeChunk
+	newChunk := NewTenantStatusTimeChunk(endTime)
+	g.activeChunk = newChunk
+	g.mu.Unlock()
+	g.LogChunks()
+}
+
+func (g *GlobalStatus) LogChunks() {
+	for tsidx, chunk := range g.timeChunks {
+		log.Debug().Time("tsidx", tsidx).Time("start time", chunk.startTime).Time("end time", chunk.endTime).Int("tenants length", len(chunk.tenants)).Msg("chunk metadata")
+		for tenantName, tenantStatus := range chunk.tenants {
+			log.Debug().Str("tenant name", tenantName).Int64("readBytes", tenantStatus.readBytes).Int64("writeBytes", tenantStatus.writeBytes).Msg("Bytes for tenant")
+			log.Debug().Str("tenant name", tenantName).Int64("writeUnits", tenantStatus.writeUnits).Int64("readUnits", tenantStatus.readUnits).Msg("Units for tenant")
+		}
+	}
+}
+
+func (g *GlobalStatus) GetActiveChunk() *TenantStatusTimeChunk {
+	return g.activeChunk
+}
+
+func (g *GlobalStatus) EnsureTenantForActiveChunk(tenantName string) {
+	_, ok := g.activeChunk.tenants[tenantName]
+	if !ok {
+		g.activeChunk.tenants[tenantName] = NewTenantStatus()
+	}
+}
+
+func (g *GlobalStatus) RecordRequestToActiveChunk(r *RequestStatus, tenantName string) {
+	log.Debug().Msg("Recording request to active chunk")
+	writeUnits := getUnitsFromBytes(r.writeBytes, config.WriteUnitSize)
+	readUnits := getUnitsFromBytes(r.readBytes, config.ReadUnitSize)
+	g.mu.Lock()
+	g.EnsureTenantForActiveChunk(tenantName)
+	log.Debug().Int64("writeBytes", r.writeBytes).Str("tenantName", tenantName).Msg("Recording write bytes")
+	g.activeChunk.tenants[tenantName].writeBytes += r.writeBytes
+	log.Debug().Int64("readBytes", r.readBytes).Str("tenantName", tenantName).Msg("Recording read bytes")
+	g.activeChunk.tenants[tenantName].readBytes += r.readBytes
+	log.Debug().Int64("writeUnits", writeUnits).Str("tenantName", tenantName).Msg("Recording write units")
+	g.activeChunk.tenants[tenantName].writeUnits += writeUnits
+	log.Debug().Int64("readUnits", readUnits).Str("tenantName", tenantName).Msg("Recording read units")
+	g.activeChunk.tenants[tenantName].readUnits += readUnits
+	g.mu.Unlock()
+}
+
+func (g *GlobalStatus) GetChunks() map[time.Time]*TenantStatusTimeChunk {
```

Comment:
Do we need this method?

Reply:
We need something for the integration to read the data.

Reply:
With moving to Flush, this become obsolete.